workflows/offline_ml_diags/offline_ml_diags/compute_diags.py

```diff
@@ -319,103 +256,82 @@ def _get_transect(ds_snapshot: xr.Dataset, grid: xr.Dataset, variables: Sequence
-    config.model_path = args.model_path
+
+def main(args):
+    logger.info("Starting diagnostics routine.")
+
+    with fsspec.open(args.data_yaml, "r") as f:
+        as_dict = yaml.safe_load(f)
+    config = loaders.BatchesLoader.from_dict(as_dict)
 
     logger.info("Reading grid...")
     if not args.grid:
         # By default, read the appropriate resolution grid from vcm.catalog
-        res = config.batch_kwargs.get("res", "c48")
-        grid = load_grid_info(res)
+        grid = load_grid_info(args.grid_resolution)
     else:
         with fsspec.open(args.grid, "rb") as f:
             grid = xr.open_dataset(f).load()
 
-    variables = list(
-        set(config.input_variables + config.output_variables + ADDITIONAL_VARS)
-    )
-
     logger.info("Opening ML model")
     model = fv3fit.load(args.model_path)
-    pred_mapper = _get_prediction_mapper(config, variables, model, grid)
-
-    # Use appropriate times if options --timesteps-file or --timesteps-n-samples given
-    if args.timesteps_file:
-        with open(args.timesteps_file, "r") as f:
-            timesteps = yaml.safe_load(f)
-        if args.timesteps_n_samples:
-            random.seed(0)
-            timesteps = random.sample(sorted(timesteps), args.timesteps_n_samples)
-        logger.info(f"Using timesteps from --timesteps-file: {timesteps}")
-        config.timesteps_source = "timesteps_file"
-    elif args.timesteps_n_samples:
-        # Sample times outside training range and use as test set.
-        train_timesteps = config.batch_kwargs.pop("timesteps", [])
-        timesteps = sample_outside_train_range(
-            list(pred_mapper), train_timesteps, args.timesteps_n_samples
-        )
-        logger.info(
-            f"Using timesteps sampled from outside config timestep range: {timesteps}"
-        )
-        config.timesteps_source = "sampled_outside_input_config"
-    else:
-        try:
-            timesteps = config.batch_kwargs["timesteps"]
-            logger.info(f"Using timesteps given in config file: {timesteps}")
-            config.timesteps_source = "input_config"
-        except KeyError:
-            timesteps = list(pred_mapper)
-            logger.info(f"Using all timesteps available from mapper: {timesteps}")
-            config.timesteps_source = "all_mapper_times"
-
-    # Updates timesteps so that the saved config reflects the timesteps used.
-    config.batch_kwargs["timesteps"] = timesteps
-
-    # write out config used to generate diagnostics, including model path
-    config.dump(args.output_path, filename="config.yaml")
-
-    batch_kwargs = dissoc(
-        config.batch_kwargs, "mapping_function", "mapping_kwargs", "timesteps"
-    )
-    batches = loaders.batches.batches_from_mapper(
-        pred_mapper, variables, timesteps=timesteps, **batch_kwargs,
-    )
+    model_variables = list(set(model.input_variables + model.output_variables + [DELP]))
+    all_variables = list(set(model_variables + ADDITIONAL_VARS))
```

Comment:
A DerivedModel will automatically (thanks to @AnnaKwa) include the additional variables in its `input_variables`. The model itself is a black box that turns inputs into outputs, and deriving output variables using something other than "machine learning" but inside a Model is still just another part of that black box.

Reply:
The term "additional_variables" is a bit confusing now that there's multiple scenarios in which "additional" variables are needed. (a) if needed for a derived output variable, the `DerivedModel` will handle this automatically now and add these to `DerivedModel.input_variables` (the set of inputs actually used for ML prediction are kept as `DerivedModel._base_model.input_variables`) (b) needed for offline diags computation, ex. pressure level thickeness, those are configured here.

Reply:
OK thanks, though I'm still confused as to whether a derived model's 'derived' output will be evaluated in the offline diags. The current docs say no: http://vulcanclimatemodeling.com/docs/fv3fit/derived_model.html, but it sounds like there's been a change such that the answer is actually yes.

Reply:
Good catch, I think those docs are out of date. This should now be documented in the TrainingConfig, since you no longer need to create that model manually. I'll expand the TrainingConfig doc a little and delete that page.

This also reminds me i need to update the docs to reference the new config objects, they still reference ModelTrainingConfig.

Reply:
Thanks, good catch, those docs are out of date. The 'derived' outputs will be evaluated in the offline diags now.